coordinator/src/main/java/org/apache/uniffle/coordinator/AppBalanceSelectStorageStrategy.java

```diff
@@ -39,106 +45,127 @@
 public class AppBalanceSelectStorageStrategy implements SelectStorageStrategy {
 
   private static final Logger LOG = LoggerFactory.getLogger(AppBalanceSelectStorageStrategy.class);
-  /**
-   * store appId -> remote path to make sure all shuffle data of the same application
-   * will be written to the same remote storage
-   */
-  private final Map<String, RemoteStorageInfo> appIdToRemoteStorageInfo;
   /**
    * store remote path -> application count for assignment strategy
    */
-  private final Map<String, RankValue> remoteStoragePathCounter;
+  private final Map<String, RankValue> remoteStoragePathRankValue;
+  private final Map<String, RemoteStorageInfo> appIdToRemoteStorageInfo;
   private final Map<String, RemoteStorageInfo> availableRemoteStorageInfo;
-
-  public AppBalanceSelectStorageStrategy() {
-    this.appIdToRemoteStorageInfo = Maps.newConcurrentMap();
-    this.remoteStoragePathCounter = Maps.newConcurrentMap();
-    this.availableRemoteStorageInfo = Maps.newHashMap();
+  private final Configuration hdfsConf;
+  private final int fileSize;
+  private final int readAndWriteTimes;
+  private boolean remotePathIsHealthy = true;
```

Comment:
What does this variable `remotePathIsHealthy`  mean?
This class is a strategy. I can't get the meaning of this varibale.

Reply:
This variable is used to determine whether the path is normal, and to pass the parameter into the method `sortPathByRankValue`.

Reply:
> This variable is used to determine whether the path is normal, and to pass the parameter into the method `sortPathByRankValue`.

One strategy should have multiple paths, every path is healthy, the variable is true?

Reply:
I know that there are many paths, but this attribute is applied to each path. The healthy path is processed by `sortPathByRankValue` according to `remotePathIsHealthy` as true, and corresponding logic is processed when it is false.

Reply:
You can look at the logic of `sortPathByRankValue`. When we judge by this parameter, it is a normal path, and then deal with it accordingly.

Reply:
Is the variable necessary to become member field?

Reply:
Updated.